docs/static/persistent-queues.asciidoc

```diff
@@ -1,44 +1,245 @@
+If you optimize for performance, you may increase your risk of losing data.
+If you optimize for data protection, you may impact performance. 
+
+[[pq-size]]
+===== Queue size
+You can control queue size with the `queue.max_events` and  `queue.max_bytes` settings.
+If both settings are specified, Logstash uses whichever criteria is reached
+first. 
+See <<backpressure-persistent-queue>> for behavior when queue limits are
+reached.
+
+[[pq-lower-max_bytes]]
+===== Smaller queue size
+If you are using persistent queues to protect against data loss, but don't
+require much buffering, you can set `queue.max_bytes` to a smaller value.
+A smaller value may produce smaller queues and improves queue performance.
+
+*Sample configuration*
+
+[source, yaml]
+-----
+queue.type: persisted
+queue.max_bytes: 10mb 
+-----
+
+[[pq-fewer-checkpoints]]
+===== Fewer checkpoints
+
+Setting `queue.checkpoint.writes` and `queue.checkpoint.acks` to `0` may
+yield maximum performance, but may have potential impact on durability.
+
+In a situation where Logstash is terminated or there is a hardware-level
+failure, any data that has not been checkpointed, is lost. 
+See <<durability-persistent-queues>> to better understand the trade-offs.
+
+[[troubleshooting-pqs]]
+==== Troubleshooting persistent queues
+
+Symptoms of persistent queue problems include {ls} or one or more pipelines not starting successfully, accompanied by an error message similar to this one.
+
+```
+message=>"java.io.IOException: Page file size is too small to hold elements
+```
+
+This error indicates that the head page (the oldest in a folder and the one with lowest page id) has a size < 18 bytes, the size of a page header.
+
+
+To resolve the issue: 
+
+. Identify the queue (or queues) that may be corrupt.
+. Stop Logstash.
+. Run `pqrepair <path>` for each of the corrupted queues.
+
+[[pqcheck]]
+===== `pqcheck` utility
+
+Run the `pqcheck` utility to identify which persistent queue--or queues--have been corrupted. 
+
+From LOGSTASH_HOME, run:
+
+[source,txt]
+-----
+bin/pqcheck <queue_folder>
+-----
 
-The following are problems not solved by the persistent queue feature:
+where `<queue_folder>` is the fully qualified path to the persistent queue location.
 
-* Input plugins that do not use a request-response protocol cannot be protected from data loss. For example: tcp, udp, zeromq push+pull, and many other inputs do not have a mechanism to acknowledge receipt to the sender. Plugins such as beats and http, which *do* have an acknowledgement capability, are well protected by this queue.
-* It does not handle permanent machine failures such as disk corruption, disk failure, and machine loss. The data persisted to disk is not replicated.
+The `pqcheck utility` reads through the checkpoint files in the given directory and outputs information about the current state of those files. 
+The utility outputs this information for each checkpoint file:
```

Comment:
Do you think it makes sense to present this as a fully annotated line from a `pqcheck` run - we do so partially later in the doc when we mention a healthy PQ and a corrupted PQ, but maybe this would be easier to correlate/read if it was presented that way?